src/BaseElement.js

```diff
@@ -302,40 +305,40 @@ class BaseElement extends HTMLElement {
 		}
 	}
 
-	/**
-	 * Context Properties to issue Context Requests and to pull contextual properties into the elements scope
-	 * @return {{}}
-	 */
-	context() {
-		return {};
-	}
-
 	/**
 	 * Defines context on the element based on keys from this.context()
 	 */
 	defineContext() {
-		Object.entries(this.context()).forEach(([key, value]) => {
-			this.dispatch('request-context', { [key]: value }, true);
-		});
 		// define context provider
-		if (this._options.provideContext) {
+		if (Object.values(this._options.propertyOptions).find((option) => option.provideContext === true)) {
+			// at least one property that provides Context
 			this.addEventListener('request-context', this.onRequestContext);
 		}
 	}
 
+	doRequestContext(property, callback) {
+		this.dispatch('request-context', { [property]: callback }, true);
+	}
+
+	/**
+	 * catch context requests and provide data if there is some
+	 * */
 	onRequestContext(event) {
 		const properties = this.properties();
 
 		Object.entries(event.detail ?? {}).forEach(([key, callback]) => {
-			if (properties.hasOwnProperty(key)) {
+			if (this._options.propertyOptions[key]?.provideContext === true && properties.hasOwnProperty(key)) {
 				// found it, provide it
 				event.stopPropagation();
 				if (typeof callback === 'function') {
 					// call function with context value
 					callback(properties[key]);
 				} else {
-					// auto define as prop
```

Comment:
@quarkus Isn't the callback in this case the value allready? Can't we just set callback = property; and be done?
And let the subscribing be done in the setter on the consumer

Reply:
check.